specs/AppNotifications/AppNotifications-spec.md

```diff
@@ -543,6 +543,367 @@ namespace Microsoft.Windows.AppNotifications
+        // For Packaged apps, the COM server is defined in the manifest. The Process calling Register() and the process defined as the COM server are required to be the same.
+        // For Unpackaged apps, the caller process will be registered as the COM server. And assets like displayname and icon will be gleaned from Shell and registered as well.
+        void Register();
+
+        // Unpackaged Apps can call this API to register custom displayname and icon for AppNotifications and register themselves as a COM server.
+        void Register(String displayName, Windows.Foundation.Uri iconUri);
+
+        // Unregisters the COM Service so that a subsequent activation will launch a new process
+        void Unregister();
+
+        // Cleans up all Registration related data for toasts. After this, toasts will not work until Register() is called again
+        void UnregisterAll();
+
+        // Event handler for Notification Activations
+        event Windows.Foundation.TypedEventHandler<AppNotificationManager, AppNotificationActivatedEventArgs> NotificationInvoked;
+
+        // Displays the Notification in Action Center
+        void Show(AppNotification notification);
+
+        // Updates the Notification for a Progress related operation using Tag and Group
+        Windows.Foundation.IAsyncOperation<AppNotificationProgressResult> UpdateAsync(AppNotificationProgressData data, String tag, String group);
+
+        // Updates the Notification for a Progress related operation using Tag
+        Windows.Foundation.IAsyncOperation<AppNotificationProgressResult> UpdateAsync(AppNotificationProgressData data, String tag);
+
+        // Get the Notification Setting status for the app
+        AppNotificationSetting Setting { get; };
+
+        // Removes a specific Notification with a specific NotificationIdentifier from Action Centre
+        Windows.Foundation.IAsyncAction RemoveByIdAsync(UInt32 notificationId);
+
+        // Removes a Notification having a specific tag
+        Windows.Foundation.IAsyncAction RemoveByTagAsync(String tag);
+
+        // Removes a Notification having a specific tag and group
+        Windows.Foundation.IAsyncAction RemoveByTagAndGroupAsync(String tag, String group);
+
+        // Remove all Notifications for a specific group
+        Windows.Foundation.IAsyncAction RemoveByGroupAsync(String group);
+
+        // Removes all the Notifications for the App from Action Centre
+        Windows.Foundation.IAsyncAction RemoveAllAsync();
+
+        // Gets all the Notifications for the App from Action Centre
+        Windows.Foundation.IAsyncOperation<Windows.Foundation.Collections.IVector<AppNotification> > GetAllAsync();
+    }
+}
+  ```
+#### AppNotificationBuilder
+  ```c#
+  namespace Microsoft.Windows.AppNotifications.Builder
+{
+    enum AppNotificationButtonSettingType
+    {
+        none,
+        VideoCall,
+        AudioCall,
+    };
+
+    runtimeclass AppNotificationButton
```

Comment:
Shouldn't the runtime classes have an apicontract?
Ideally, we are supposed to add the new contract in specs/WinRT/WinRTAPIContracts.md as well.

Reply:
contracts has been added in actual code, but not here.. the PR for push actual code changes will have those details.